app/controllers/concerns/blacklight/controller.rb

```diff
@@ -52,134 +52,134 @@ def default_catalog_controller
+  end
 
-    ##
-    # Determine whether to render the bookmarks control
-    # (Needs to be available globally, as it is used in the navbar)
-    def render_bookmarks_control?
-      has_user_authentication_provider? && current_or_guest_user.present?
-    end
+  ##
+  # Determine whether to render the bookmarks control
+  # (Needs to be available globally, as it is used in the navbar)
+  def render_bookmarks_control?
+    has_user_authentication_provider? && current_or_guest_user.present?
+  end
 
-    ##
-    # Determine whether to render the saved searches link
-    # (Needs to be available globally, as it is used in the navbar)
-    def render_saved_searches?
-      has_user_authentication_provider? && current_user
-    end
+  ##
+  # Determine whether to render the saved searches link
+  # (Needs to be available globally, as it is used in the navbar)
+  def render_saved_searches?
+    has_user_authentication_provider? && current_user
+  end
 
-    # @return [Blacklight::SearchState] a memoized instance of the parameter state.
-    def search_state
-      @search_state ||= search_state_class.new(params, blacklight_config)
-    end
+  # @return [Blacklight::SearchState] a memoized instance of the parameter state.
+  def search_state
+    @search_state ||= search_state_class.new(params, blacklight_config)
+  end
 
-    # Default route to the search action (used e.g. in global partials). Override this method
-    # in a controller or in your ApplicationController to introduce custom logic for choosing
-    # which action the search form should use
-    def search_action_url options = {}
-      # Rails 4.2 deprecated url helpers accepting string keys for 'controller' or 'action'
-      search_catalog_url(options.except(:controller, :action))
+  # Default route to the search action (used e.g. in global partials). Override this method
+  # in a controller or in your ApplicationController to introduce custom logic for choosing
+  # which action the search form should use
+  def search_action_url options = {}
+    # Rails 4.2 deprecated url helpers accepting string keys for 'controller' or 'action'
+    search_catalog_url(options.except(:controller, :action))
+  end
+
+  def search_action_path *args
+    if args.first.is_a? Hash
+      args.first[:only_path] = true
     end
 
-    def search_action_path *args
-      if args.first.is_a? Hash
-        args.first[:only_path] = true
-      end
+    search_action_url(*args)
+  end
 
-      search_action_url(*args)
-    end
+  def search_facet_url options = {}
+    opts = search_state.to_h.merge(action: "facet").merge(options).except(:page)
+    url_for opts
+  end
+  deprecation_deprecate search_facet_url: 'Use search_facet_path instead.'
 
-    def search_facet_url options = {}
-      opts = search_state.to_h.merge(action: "facet").merge(options).except(:page)
-      url_for opts
+  def search_facet_path(options = {})
```

Comment:
@atz what do you mean by semantic  changes?  I think the diff is perhaps misleading.   Take a look below.